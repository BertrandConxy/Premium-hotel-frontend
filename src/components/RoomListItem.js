import { React, useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { deleteRoom } from '../redux/rooms/rooms';
import './roomDetails.css';

const RoomListItem = ({ name, id }) => {
  const [deleted, setDeleted] = useState(false);
  const dispatch = useDispatch();

  const handleClick = (id) => {
    dispatch(deleteRoom(id));
    setDeleted(true);
  };
  return (
    <div className="d-flex justify-content-between align-items-center py-1 px-3 shadow rounded-3">
      <h3 className="fw-bold">{name}</h3>
      <button
        type="button"
        className={deleted ? 'btn btn-danger disabled' : 'buttonConfig upperClass'}
        onClick={() => handleClick(id)}
      >
        <h3>{deleted ? 'Removed' : 'Delete'}</h3>
      </button>
    </div>
  );
};

RoomListItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

export default RoomListItem;
