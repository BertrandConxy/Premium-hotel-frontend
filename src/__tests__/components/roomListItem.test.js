import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import RoomListItem from '../../components/RoomListItem';
import store from '../../redux/configureStore';

const name = 'Appart 1';
const id = 1;
const RoomListItemProvider = () => (
  <Provider store={store}>
    <RoomListItem name={name} id={id} />
  </Provider>
);

describe('render component', () => {
  it('renders the room picture and button', () => {
    render(<RoomListItemProvider />);
    const roomName = screen.getByText('Appart 1');
    const roomButton = screen.getByText('Delete');
    expect(roomName).toBeInTheDocument();
    expect(roomButton).toBeInTheDocument();
  });
});

describe('Snapshot test', () => {
  it('looks the same as the snapshot', () => {
    const component = renderer.create(<RoomListItemProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
