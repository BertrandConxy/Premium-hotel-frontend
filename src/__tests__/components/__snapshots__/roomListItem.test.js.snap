// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<div
  className="d-flex justify-content-between align-items-center py-1 px-3 shadow rounded-3"
>
  <h3
    className="fw-bold"
  >
    Appart 1
  </h3>
  <button
    className="buttonConfig upperClass"
    onClick={[Function]}
    type="button"
  >
    <h3>
      Delete
    </h3>
  </button>
</div>
`;
